python/pyspark/sql/tests/test_udtf.py

```diff
@@ -1064,6 +1074,71 @@ def eval(self, row: Row):
         func = udtf(TestUDTF, returnType="a: int")
         return func
 
+    def test_df_asTable_chaining_methods(self):
+        class TestUDTF:
+            def eval(self, row: Row):
+                yield row["key"], row["value"]
+
+            def terminate(self):
+                if False:
+                    yield
```

Comment:
What is this for?

Reply:
That is necessary with "partitionBy" otherwise we will hit

```
AttributeError: 'TestUDTF' object has no attribute 'terminate'
```

because of https://github.com/apache/spark/blob/master/python/pyspark/worker.py#L1052.

I will file a fix separately.

Reply:
https://github.com/apache/spark/pull/49299

Reply:
Now that #49299 was merged, we can remove this?